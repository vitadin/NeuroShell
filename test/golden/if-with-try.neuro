%% Integration testing: \if command with \try command
%% Tests interaction between conditional execution and error handling
\set[_echo_command="true"]

%% Test \if inside \try - successful condition
\try \if[condition=true] \echo "If inside try - success"
\get[_status]
\get[_error]

%% Test \if inside \try - false condition (should not execute)
\try \if[condition=false] \echo "This should not appear"
\get[_status]
\get[_error]

%% Test \if inside \try - malformed if command
\try \if \echo "This should fail"
\get[_status]
\get[_error]

%% Test \try inside \if - successful condition with successful try
\if[condition=true] \try \echo "Try inside if - success"
\get[_status]
\get[_output]

%% Test \try inside \if - successful condition with failing try
\if[condition=true] \try \bash ls /nonexistent_directory_test
\get[_status]
\get[_error]

%% Test \try inside \if - false condition (try should not execute)
\if[condition=false] \try \echo "This should not execute"
\get[_status]

%% Test using try status variables in if conditions
\try \bash echo "success"
\if[condition=${_status}] \echo "Command failed (status != 0)"
\if[condition=${_output}] \echo "Command produced output: ${_output}"

%% Test conditional error handling based on try results
\try \bash ls /does_not_exist
\if[condition=${_status}] \echo "Previous command failed with status: ${_status}"
\if[condition=${_error}] \echo "Error message exists: ${_error}"

%% Test successful command status
\try \set[test_var="success"]
\if[condition=${_status}] \echo "Set command failed unexpectedly"
\get[test_var]

%% Test chaining try and if for robust error handling
\try \bash echo "test output"
\if[condition=${_output}] \set[output_captured="true"]
\get[output_captured]

%% Test nested scenarios - if containing try containing commands
\if[condition=true] \try \set[nested_var="nested_success"]
\get[nested_var]
\get[_status]

%% Test using if conditions based on previous if results
\if[condition=true] \set[first_result="true"]
\if[condition=${first_result}] \echo "Chained if conditions work"

%% Test error propagation through if/try combinations
\set[should_fail="false"]
\if[condition=${should_fail}] \try \bash this_command_does_not_exist
\get[_status]

%% Test complex conditional logic with try/if
\try \bash echo "conditional test"
\set[has_output="true"]
\if[condition=${has_output}] \echo "Output detected: ${_output}"
\if[condition=${_status}] \echo "Command had error status"

%% Test that #if_result is preserved across try operations
\if[condition=true] \set[dummy="test"]
\get[#if_result]
\try \bash echo "test"
\get[#if_result]

%% Test if condition based on system error state
\try \bash false
\if[condition=${_status}] \set[error_detected="true"]
\get[error_detected]

%% Test successful path through if/try combination
\try \bash true  
\if[condition=${_status}] \echo "Unexpected error"
\if[condition=${_output}] \echo "Unexpected output"
\echo "Success path completed"