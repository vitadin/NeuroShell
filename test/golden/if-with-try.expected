Setting _echo_command = true
%%> "\\try \\if[condition=true] \\echo \"If inside try - success\""
%%> "\\if[condition=true] \\echo \"If inside try - success\""
%%> "\\echo \"If inside try - success\""
"If inside try - success"
%%> "\\get[_status]"
_status = 0
%%> "\\get[_error]"
_error = 
%%> "\\try \\if[condition=false] \\echo \"This should not appear\""
%%> "\\if[condition=false] \\echo \"This should not appear\""
%%> "\\get[_status]"
_status = 0
%%> "\\get[_error]"
_error = 
%%> "\\try \\if \\echo \"This should fail\""
%%> "\\if \\echo \"This should fail\""
%%> "\\get[_status]"
_status = 1
%%> "\\get[_error]"
_error = condition parameter is required
%%> "\\if[condition=true] \\try \\echo \"Try inside if - success\""
%%> "\\try \\echo \"Try inside if - success\""
%%> "\\echo \"Try inside if - success\""
"Try inside if - success"
%%> "\\get[_status]"
_status = 0
%%> "\\get[_output]"
_output = "Try inside if - success"
%%> "\\if[condition=true] \\try \\bash ls /nonexistent_directory_test"
%%> "\\try \\bash ls /nonexistent_directory_test"
%%> "\\bash ls /nonexistent_directory_test"
Error: <ls_error_macos>
<non_zero_exit_status>
%%> "\\get[_status]"
_status = 1
%%> "\\get[_error]"
_error = <ls_error_macos>
%%> "\\if[condition=false] \\try \\echo \"This should not execute\""
%%> "\\get[_status]"
_status = 0
%%> "\\try \\bash echo \"success\""
%%> "\\bash echo \"success\""
success
%%> "\\if[condition=${_status}] \\echo \"Command failed (status != 0)\""
%%> "\\if[condition=${_output}] \\echo \"Command produced output: ${_output}\""
%%> "\\echo \"Command produced output: success\""
"Command produced output: success"
%%> "\\try \\bash ls /does_not_exist"
%%> "\\bash ls /does_not_exist"
Error: <ls_error_macos>
<non_zero_exit_status>
%%> "\\if[condition=${_status}] \\echo \"Previous command failed with status: ${_status}\""
%%> "\\echo \"Previous command failed with status: 1\""
"Previous command failed with status: 1"
%%> "\\if[condition=${_error}] \\echo \"Error message exists: ${_error}\""
%%> "\\try \\set[test_var=\"success\"]"
%%> "\\set[test_var=\"success\"]"
Setting test_var = success
%%> "\\if[condition=${_status}] \\echo \"Set command failed unexpectedly\""
%%> "\\get[test_var]"
test_var = success
%%> "\\try \\bash echo \"test output\""
%%> "\\bash echo \"test output\""
test output
%%> "\\if[condition=${_output}] \\set[output_captured=\"true\"]"
%%> "\\set[output_captured=\"true\"]"
Setting output_captured = true
%%> "\\get[output_captured]"
output_captured = true
%%> "\\if[condition=true] \\try \\set[nested_var=\"nested_success\"]"
%%> "\\try \\set[nested_var=\"nested_success\"]"
%%> "\\set[nested_var=\"nested_success\"]"
Setting nested_var = nested_success
%%> "\\get[nested_var]"
nested_var = nested_success
%%> "\\get[_status]"
_status = 0
%%> "\\if[condition=true] \\set[first_result=\"true\"]"
%%> "\\set[first_result=\"true\"]"
Setting first_result = true
%%> "\\if[condition=${first_result}] \\echo \"Chained if conditions work\""
%%> "\\echo \"Chained if conditions work\""
"Chained if conditions work"
%%> "\\set[should_fail=\"false\"]"
Setting should_fail = false
%%> "\\if[condition=${should_fail}] \\try \\bash this_command_does_not_exist"
%%> "\\get[_status]"
_status = 0
%%> "\\try \\bash echo \"conditional test\""
%%> "\\bash echo \"conditional test\""
conditional test
%%> "\\set[has_output=\"true\"]"
Setting has_output = true
%%> "\\if[condition=${has_output}] \\echo \"Output detected: ${_output}\""
%%> "\\echo \"Output detected: conditional test\""
"Output detected: conditional test"
%%> "\\if[condition=${_status}] \\echo \"Command had error status\""
%%> "\\if[condition=true] \\set[dummy=\"test\"]"
%%> "\\set[dummy=\"test\"]"
Setting dummy = test
%%> "\\get[#if_result]"
#if_result = true
%%> "\\try \\bash echo \"test\""
%%> "\\bash echo \"test\""
test
%%> "\\get[#if_result]"
#if_result = true
%%> "\\try \\bash false"
%%> "\\bash false"
<non_zero_exit_status>
%%> "\\if[condition=${_status}] \\set[error_detected=\"true\"]"
%%> "\\set[error_detected=\"true\"]"
Setting error_detected = true
%%> "\\get[error_detected]"
error_detected = true
%%> "\\try \\bash true"
%%> "\\bash true"
%%> "\\if[condition=${_status}] \\echo \"Unexpected error\""
%%> "\\if[condition=${_output}] \\echo \"Unexpected output\""
%%> "\\echo \"Success path completed\""
"Success path completed"